src/natcap/invest/utils.py

```diff
@@ -43,6 +44,173 @@
 # leaves Projected CRS alone
 DEFAULT_OSR_AXIS_MAPPING_STRATEGY = osr.OAMS_TRADITIONAL_GIS_ORDER
 
+# the same unit registry instance should be shared across everything
+u = pint.UnitRegistry()
+u.define('dollar = [value]')  # adding a new dimension "value"
+u.define('million_dollars = 1000000 * dollar')
+u.define('currency = []')         # non-specific unit of value
+u.define('pixel = []')            # non-specific unit of area
+u.define('projection_unit = []')  # non-specific unit of length
+# used in coastal vulnerability, the DEM pixel values measure
+# elevation (length) but the specific units don't matter
+u.define('linear_unit_of_elevation = []')  # non-specific unit of length
+# add "us_survey_foot" on to the aliases because it's used in some rasters
+u.define('survey_foot = 1200 / 3937 * meter = sft = us_survey_foot')
+# Vitamin A in the crop production nutrient table is measured in IUs
+# A special unit in pharmacology that measures biologically active substances
+# May be converted to weight or volume, but conversion factors are specific
+# to the substance. I couldn't find a definition of its dimensionality.
+u.define('international_unit = [biologic_amount] = iu = IU')
+# Use u.none for unitless measurements
+u.define('none = []')
+
+AREA = {
+    "type": "vector",
+    "fields": {},
+    "geometries": {"POLYGON", "MULTIPOLYGON"}
+}
+
+PROJECTED = {
+    "validation_options": {
+        "projected": True
+    }
+}
+
+METER_PROJECTED = {
+    "validation_options": {
+        "projected": True,
+        "projection_units": u.meter
+    }
+}
+
+PROJECTED_AREA = {
+    **AREA,
+    **PROJECTED
+}
+
+METER_PROJECTED_AREA = {
+    **AREA,
+    **METER_PROJECTED
+}
+
+GT_0 = {
+    "validation_options": {
+        "expression": "value > 0"
+    }
+}
+GTE_0 = {
+    "validation_options": {
+        "expression": "value >= 0"
+    }
+}
+BETWEEN_0_AND_1 = {
+    "validation_options": {
+        "expression": "(value >= 0) & (value <= 1)"
+    }
+}
+
+DISTANCE = {
+    "type": "number",
+    "units": u.meter,
+    "validation_options": {
+        "expression": "value >= 0"
+    }
+}
+
+METER_RASTER = {
+    "type": "raster",
+    "bands": {
+        1: {
+            "type": "number",
+            "units": u.meter
+        }
+    }
+}
+
+AOI_ARG = {
+    **AREA,
+    "name": "area of interest",
+    "about": (
+        "A GDAL-supported vector file.  This AOI instructs "
+        "the model where to clip the input data and the extent "
+        "of analysis.  Users will create a polygon feature "
+        "layer that defines their area of interest.  The AOI "
+        "must intersect the Digital Elevation Model (DEM)."),
```

Comment:
You're right, this should get re-worded to be more generic. I'm thinking it should be pretty basic and then you can override it in the model if needed:
```
{
    **utils.AOI_ARG,
    "about": "custom about text..."
}
```

Reply:
in case anyone's curious: I wasn't sure that putting a new `"about"` key would always override the existing `"about"` key from the unpacked dictionary. But it seems dictionaries are evaluated in order: _"you can specify the same key multiple times in the key/datum list, and the final dictionary’s value for that key will be the last one given... Later values replace values already set by earlier key/datum pairs and earlier dictionary unpackings."_ https://docs.python.org/3/reference/expressions.html#dictionary-displays

Reply:
Thanks @emlys , I was curious!